opentelemetry-sdk/benches/batch_log_processor.rs

```diff
@@ -0,0 +1,138 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use opentelemetry::logs::LoggerProvider as _;
+use opentelemetry::logs::Severity;
+use opentelemetry::logs::{LogRecord, Logger};
+use opentelemetry_sdk::export::logs::LogExporter;
+use opentelemetry_sdk::logs::LoggerProvider;
+use opentelemetry_sdk::logs::{BatchConfigBuilder, BatchLogProcessor};
+use opentelemetry_sdk::resource::Resource;
+use opentelemetry_sdk::runtime::Tokio;
+use std::sync::{Arc, Condvar, Mutex};
+use std::time::SystemTime;
+use tokio::runtime::Runtime;
+
+#[derive(Debug)]
+struct NoopLogExporter;
+
+#[async_trait::async_trait]
+impl LogExporter for NoopLogExporter {
+    async fn export<'a>(
+        &mut self,
+        _batch: Vec<std::borrow::Cow<'a, opentelemetry_sdk::export::logs::LogData>>,
+    ) -> opentelemetry::logs::LogResult<()> {
+        Ok(())
+    }
+
+    fn shutdown(&mut self) {}
+
+    fn set_resource(&mut self, _resource: &Resource) {}
+}
+
+#[derive(Debug)]
+struct MockLogExporter {
+    export_signal: Arc<(Mutex<bool>, Condvar)>,
+}
+
+#[async_trait::async_trait]
+impl LogExporter for MockLogExporter {
+    async fn export<'a>(
+        &mut self,
+        _batch: Vec<std::borrow::Cow<'a, opentelemetry_sdk::export::logs::LogData>>,
+    ) -> opentelemetry::logs::LogResult<()> {
+        let (lock, cvar) = &*self.export_signal;
+        let mut exported = lock.lock().unwrap();
+        *exported = true;
+        cvar.notify_all(); // Notify that the batch has been exported
+        Ok(())
+    }
+
+    fn shutdown(&mut self) {}
+
+    fn set_resource(&mut self, _resource: &Resource) {}
+}
+
+fn benchmark_batch_log_processor(c: &mut Criterion) {
+    // Benchmark 1: Measure time to emit and flush a log record
```

Comment:
can you make a benchmark to know the hot path cost alone? Benchmark 1 is doing flush and benchmark 2 is doing many items till export gets triggered, so neither is matching the cost typically expected by the user which is the cost of creating logrecord, calling emit, passing to processor, cloning by batch processor, queueing to the channel.

Reply:
Yes, @utpilla also suggested same :). Have done now. Thanks.